### PR TITLE
[release-4.15]: OCPBUGS-35076: Release Fix CVE go-jose

### PR DESCRIPTION
**Pull request by marioferh**
fix CVE-2024-28180 
fix of https://github.com/openshift/telemeter/pull/531

bump go-jose to fix CVE and remove go-jose versions with vulnerability 

go-jose v3.0.3 
oidc to v3.9.0

### CODE REVIEW

**Review by marioferh — COMMENTED** [review]
CVE

File: pkg/authorize/jwt/claims.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package jwt
 import (
 	"time"
 
-	"gopkg.in/square/go-jose.v2/jwt"
+	"github.com/go-jose/go-jose/v3/jwt"
```
Comment (machine424) [inline pkg/authorize/jwt/claims.go:6]: if we could continue using a v2 version, it'd be great.
Have you tried with `gopkg.in/go-jose/go-jose.v2`?

**Comment by machine424** [comment]
Could you add more details about the change, to the commit message or the description? Thanks.

File: pkg/authorize/jwt/claims.go (hunk 1 above)
Comment (marioferh) [reply pkg/authorize/jwt/claims.go]: done, I had issues in with v2 and I chose v3 in master but we can fix it here with v2.

**Comment by simonpasquier** [comment]
/assign @thibaultmg 

Thibault, can you have a look at this PR? Since it bumps a dependency used by the server side, I'd prefer if someone from RHOBS could double-check.

**Comment by machine424** [comment]
cc @juzhao, this and its backport https://github.com/openshift/telemeter/pull/535 need your eyes ;) thanks.

**Pull request merged**